app/services/create_order_service.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+class CreateOrderService
+  attr_reader :errors, :order, :reward, :address
+
+  def initialize(params:, employee:)
+    @order = Order.new(reward: @reward, reward_snapshot: @reward, address_snapshot: @address, employee: @employee)
+    @reward = Reward.find(params[:reward_id])
+    @employee = employee
+
+    @address = @employee.address || Address.new(employee: @employee)
+    @street = params.dig(:address, :street)
+    @city = params.dig(:address, :city)
+    @postcode = params.dig(:address, :postcode)
+
+    @errors = []
+  end
+
+  def call
+    return false unless items_in_stock?
+    return false unless sufficient_funds?
+
+    ActiveRecord::Base.transaction do
+      update_address if @reward.post?
```

Comment:
Looks like we have `if @reward.post?` check in many places, which can mean that `post` reward is significantly different than online reward. This means that we could write more specialised classes/services for them instead of having one general service. Probably strategy design pattern could be used here, https://www.sihui.io/design-pattern-strategy/